src/Temporalio/Converters/IPayloadConverter.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using Temporalio.Api.Common.V1;
+
+namespace Temporalio.Converters
+{
+    /// <summary>
+    /// Representation of a converter from a value to/from a payload.
+    /// </summary>
+    /// <remarks>
+    /// This converter should be deterministic since it is used for workflows. For the same reason,
+    /// this converter should be immediate and avoid any network calls or any asynchronous/slow code
+    /// paths.
+    /// </remarks>
+    /// <seealso cref="PayloadConverter" />
+    public interface IPayloadConverter
```

Comment:
I brought this up on internal Slack. Here's the thought progression that happened:

Go and Java had single and multi-payload methods for its data converter. But we end up leveraging multiple for remote data converters which made remote seem wise. We chose multiple in Python because type hints were optional and it matched Go.

In .NET I originally had collection to collection here with optional types. But it became clear that .NET doesn't allow optional types for JSON conversion. And really, .NET doesn't allow optional types anywhere. So like Go and Java we are going to require types. Once we made that decision, it became clear in the interest of generics that one at a time was clearest (see `ConverterExtensions` for the `ToValue` I expect most will use). After discussion, it also became clear that the arity was never allowed to change and since we don't have remote data converters or any need to have multiple at a time, we just went to one for code clarity.

I think we'd do one at a time in all SDKs if we had it to do over, but at least TypeScript, Ruby, and .NET will.